### DISCUSSION TITLE
autoconf on windows ignores autotool macros provided by ports

### DISCUSSION THREAD

**Discussion by wrobelda** [microsoft/vcpkg#20616] [Q&A]
**Describe the bug**
I am working on getting `gwenhywfar` port to build on Windows, which requires AUTOCONFIG option. The upstream `configure.ac` uses autotool macros provided by the lib's dependencies. ~~This works fine on Linux/macOS, but not on Windows. ~~ Apparently this fails on Linux, too. And it only worked on my macOS because of the libgcrypt package installed by brew.  

**Environment**
- OS: All

**To Reproduce**

1. Checkout https://github.com/microsoft/vcpkg/pull/20572
1. ./vcpkg install libgwenhywfar
2. See error

**Expected behavior**
Autoconfig ~~on Windows~~ should include the macros provided by ports

**Failure logs** 
```
configure.ac:681: error: possibly undefined macro: AM_PATH_LIBGCRYPT
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
```

**Comment by Neumann-A** [comment]
> uses autotool macros provided by the lib's dependencies

a) the macros need to be installed and autoconfig needs to be setup by the port to consume it.
b) the only way vcpkg could provide a consistent m4 behavior is to supply all of the m4 scripts itself instead of using msys

So according to the file list you have/need:
`libgcrypt:/share/libgcrypt/aclocal/libgcrypt.m4`

the solution is in #9966 add:
`set(ENV{ACLOCAL} "aclocal -I \"${CURRENT_INSTALLED_DIR}/share/libgcrypt/aclocal/\"")`
to the portfile of gwenhywfar

**Reply by wrobelda** [reply]
I have this problem now with gettext.m4, but `set(ENV{ACLOCAL} "aclocal -I \"${CURRENT_INSTALLED_DIR}/share/gettext/aclocal/\"")` does nothing. Honestly no idea what's wrong here, but I suspect since your ports were for X libraries, you didn't test this with Windows.

EDIT: appeared to be some weird x86/x64 conflict

**Reply by Neumann-A** [reply]
> Honestly no idea what's wrong here, but I suspect since your ports were for X libraries, you didn't test this with Windows.

Nope I did all the work on windows
